src/main/kotlin/org/pkl/lsp/analyzers/SyntaxAnalyzer.kt

```diff
@@ -17,21 +17,18 @@ package org.pkl.lsp.analyzers
 
 import org.pkl.lsp.ErrorMessages
 import org.pkl.lsp.Project
+import org.pkl.lsp.ast.PklError
 import org.pkl.lsp.ast.PklNode
 
 class SyntaxAnalyzer(project: Project) : Analyzer(project) {
   override fun doAnalyze(node: PklNode, diagnosticsHolder: MutableList<PklDiagnostic>): Boolean {
 
-    val del = node.checkClosingDelimiter()
-    if (del != null) {
-      val err =
-        if (del == ",") {
-          ErrorMessages.create("missingCommaSeparator")
-        } else {
-          ErrorMessages.create("missingDelimiter", del)
-        }
-      diagnosticsHolder += error(node, err)
+    if (node is PklError) {
+      diagnosticsHolder += error(node, ErrorMessages.create("invalidSyntax"))
```

Comment:
Not the _best_ error message; is there no way to reconstruct some extra details about expectations?

Reply:
The whole syntax analyzer is in a pretty poor state and should be improved. Unfortunately the only thing the error node gives is its text, but that's already shown in the client.